src/pivot-table/components/cells/DimensionTitleCell.tsx

```diff
@@ -1,38 +1,129 @@
+import Descending from "@qlik-trial/sprout/icons/react/Descending";
+import HeadCellMenu, { MenuAvailabilityFlags } from "@qlik/nebula-table-utils/lib/components/HeadCellMenu";
+import React, { useMemo, useRef } from "react";
+import type {
+  Align,
+  ChangeActivelySortedHeader,
+  ChangeSortOrder,
+  Header,
+  HeaderCell,
+  SortDirection,
+} from "../../../types/types";
+import { HEADER_TEXT_MIN_WIDTH, MENU_ICON_WIDTH, SORT_ICON_WIDTH } from "../../constants";
 import { useStyleContext } from "../../contexts/StyleProvider";
 import { getBorderStyle, textStyle } from "../shared-styles";
 
-interface LabelCellProps {
-  cell: string;
+interface DimensionTitleCellProps {
+  cell: HeaderCell;
   style: React.CSSProperties;
   isLastColumn: boolean;
+  translator: stardust.Translator;
+  changeSortOrder: ChangeSortOrder;
+  changeActivelySortedHeader: ChangeActivelySortedHeader;
 }
 
+const baseFlex: React.CSSProperties = {
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+};
+
+const labelWrapperStyle: React.CSSProperties = {
+  ...baseFlex,
+  flexDirection: "row",
+  overflow: "hidden",
+  minWidth: SORT_ICON_WIDTH + HEADER_TEXT_MIN_WIDTH,
+};
+
 const labelTextStyle: React.CSSProperties = {
   ...textStyle,
   fontWeight: "600",
-  alignSelf: "flex-end",
+  alignSelf: "center",
   flexGrow: 1,
+  paddingLeft: "8px",
+  minWidth: HEADER_TEXT_MIN_WIDTH,
 };
 
 export const testId = "title-cell";
 
-const DimensionTitleCell = ({ cell, style, isLastColumn }: LabelCellProps): JSX.Element => {
+const DimensionTitleCell = ({
+  cell,
+  style,
+  isLastColumn,
+  translator,
+  changeSortOrder,
+  changeActivelySortedHeader,
+}: DimensionTitleCellProps): JSX.Element => {
   const styleService = useStyleContext();
   const { fontSize, fontFamily } = styleService.header;
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  const isDim = cell.id !== "PSEUDO-DIM";
+
+  const headerData = useMemo<Header>(
+    () => ({
+      id: cell.id,
+      isDim,
+      fieldId: cell.fieldId,
+      qLibraryId: cell.qLibraryId,
+      label: "right",
+      headTextAlign: "right" as Align,
+      sortDirection: cell.sortDirection,
+      colIdx: cell.colIdx,
+      qReverseSort: cell.qReverseSort,
+      isActivelySorted: cell.isActivelySorted,
+    }),
+    [cell, isDim],
+  );
+
+  const sortFromMenu = async (evt: React.MouseEvent, newSortDirection: SortDirection) => {
+    evt.stopPropagation();
+    await changeSortOrder(headerData, newSortDirection);
+  };
 
   return (
     <div
-      title={cell}
+      title={cell.title}
       style={{
         ...style,
         ...getBorderStyle(true, isLastColumn, styleService.grid.border, false),
         ...styleService.header.rowTitle,
-        display: "flex",
+        padding: 0,
+        position: "relative",
+        display: "grid",
+        gridTemplateColumns: "1fr 24px",
+        gridGap: "4px",
+        minWidth: SORT_ICON_WIDTH + HEADER_TEXT_MIN_WIDTH + MENU_ICON_WIDTH,
```

Comment:
This fixes the issue where the "hamburger icon" is not aligned vertically.

Before:
![Skärmavbild 2023-09-01 kl  08 53 43](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/6fe0d6eb-9463-4173-a271-98d751dc18c7)

After:
![Skärmavbild 2023-09-01 kl  08 53 33](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/3252a543-9b1c-47a4-8089-0bcd1a113ee8)


```suggestion
        minWidth: SORT_ICON_WIDTH + HEADER_TEXT_MIN_WIDTH + MENU_ICON_WIDTH,
        alignItems: "center",
```